iib/workers/tasks/iib_static_types.py

```diff
@@ -0,0 +1,136 @@
+from typing import Dict, List, Optional, Set, TypedDict, Union
+
+from operator_manifest.operator import ImageName, OperatorCSV
+
+
+class BundleMetadata(TypedDict):
+    """Type class referencing data of bundle used for mypy checking."""
+
+    found_pullspecs: Set[ImageName]
+    operator_csvs: List[OperatorCSV]
+
+
+class IndexImageInfo(TypedDict):
+    """Type class referencing data related to index image used for mypy checking."""
+
+    arches: Set[str]
+    ocp_version: str
+    resolved_distribution_scope: str
+    resolved_from_index: Optional[str]
+
+
+class AllIndexImagesInfo(TypedDict):
+    """Type class referencing group of IndexImageInfo classes used for mypy checking."""
+
+    from_index: IndexImageInfo
+    source_from_index: IndexImageInfo
+    target_index: IndexImageInfo
+
+
+# MYPY check - should label be optional? If not total parameter can be removed.
+class PrebuildInfo(TypedDict, total=False):
+    """Type class referencing data related to preparation of request for building the image."""
+
+    arches: Set[str]
+    binary_image: str
+    binary_image_resolved: str
+    bundle_mapping: Dict[str, List[str]]
+    distribution_scope: str
+    from_index_resolved: str
+    labels: Optional[Dict[str, str]]
+    ocp_version: str
+    source_from_index_resolved: str
+    source_ocp_version: str
+    target_index_resolved: str
+    target_ocp_version: str
+
+
+class BundleImage(TypedDict):
+    """Type class referencing data related to bundle image used for mypy checking."""
+
+    bundlePath: str
+    csvName: Optional[str]
+    packageName: str
```

Comment:
Don't we need `total=False` here as well since `csvName` can be optional?

Reply:
Sure, updated. Thanks. However Optional[] does not mean that class has to have total=false, Optional means that parameter could be set to None value, and total=false means that it is possible that some of the keys will be missing.